middleware/validation.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const path = require('path')
+const fs = require('fs')
+
+/*
+  This method creates a joi vaidation schema.
+  It returns null if the schema creation fails and updates the context with an error
+*/
+const createJoiValidationSchema = (ctx) => {
+  let validations
+  const schemaObject = {}
+
+  try {
+    const resourcePath = path.resolve(__dirname, '..', 'endpoints', ctx.resourceName, 'input-validation.json')
+    const file = fs.readFileSync(resourcePath)
+    validations = JSON.parse(file)
+  } catch (error) {
+    ctx.error = error
+    return null
+  }
+
+  if (validations) {
+    Object.keys(validations).forEach(key => {
+      let rule = validations[`${key}`]
+      switch (rule.type) {
+        case 'string':
+          if (rule.required) {
+            schemaObject[`${key}`] = Joi.string().required()
+          } else if (rule.default) {
+            schemaObject[`${key}`] = Joi.string().default(rule.default)
+          } else {
+            schemaObject[`${key}`] = Joi.string()
+          } 
+          break;
+        
+        case 'number':
+          if (rule.required) {
+            schemaObject[`${key}`] = Joi.number().required()
+          } else if (rule.default) {
+            schemaObject[`${key}`] = Joi.number().default(rule.default)
+          } else {
+            schemaObject[`${key}`] = Joi.number()
+          } 
+          break;
+        
+        case 'boolean':
+          if (rule.required) {
+            schemaObject[`${key}`] = Joi.boolean().required()
+          } else if (rule.default) {
+            schemaObject[`${key}`] = Joi.boolean().default(rule.default)
+          } else {
+            schemaObject[`${key}`] = Joi.boolean()
+          }
+          break;
+        
+        case 'array':
+          if (rule.required) {
+            schemaObject[`${key}`] = Joi.array().required()
+          } else if (rule.default) {
+            schemaObject[`${key}`] = Joi.array().default(rule.default)
+          } else {
+            schemaObject[`${key}`] = Joi.array()
+          }
+          break;
+
+        default:
+
```

Comment:
For sanity, can we long a warning here if/when this case is triggered.

Reply:
fixed